src/app/core/user/user-details/user-details.component.ts

```diff
@@ -0,0 +1,282 @@
+import {
+  Component,
+  computed,
+  effect,
+  inject,
+  input,
+  output,
+  ChangeDetectionStrategy,
+  signal,
+} from "@angular/core";
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+} from "@angular/forms";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatInputModule } from "@angular/material/input";
+import { MatSelectModule } from "@angular/material/select";
+import { MatTooltipModule } from "@angular/material/tooltip";
+import { MatButtonModule } from "@angular/material/button";
+import { MatDialogModule } from "@angular/material/dialog";
+import { Role, UserAccount } from "../user-admin-service/user-account";
+import { UserAdminService } from "../user-admin-service/user-admin.service";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { DialogCloseComponent } from "../../common-components/dialog-close/dialog-close.component";
+import { AlertService } from "../../alerts/alert.service";
+import { Entity } from "../../entity/model/entity";
+
+export interface UserDetailsConfig {
+  userAccount: UserAccount | null;
+  entity: Entity | null;
+  showPasswordChange: boolean;
+  disabled: boolean;
+  editing: boolean;
+  userIsPermitted: boolean;
+  isInDialog: boolean;
+}
+
+export interface UserDetailsAction {
+  type:
+    | "formCancel"
+    | "editRequested"
+    | "closeDialog"
+    | "accountCreated"
+    | "accountUpdated";
+  data?: any;
+}
+
+/**
+ * Reusable user account details form component.
+ */
+@UntilDestroy()
+@Component({
+  selector: "app-user-details",
+  templateUrl: "./user-details.component.html",
+  styleUrls: ["./user-details.component.scss"],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [
+    ReactiveFormsModule,
+    MatFormFieldModule,
+    MatInputModule,
+    MatSelectModule,
+    MatTooltipModule,
+    MatButtonModule,
+    MatDialogModule,
+    DialogCloseComponent,
+  ],
+})
+export class UserDetailsComponent {
+  private fb = inject(FormBuilder);
+  private userAdminService = inject(UserAdminService);
+  private alertService = inject(AlertService);
+
+  config = input.required<UserDetailsConfig>();
```

Comment:
usually, we have followed a pattern to put all config properties directly as inputs into the component (not as a single config object). That makes it more visible and less nested to handle, in my opinion (and would save all the computed separate signals, because they are already input signals)